bottom/__init__.py

```diff
@@ -11,42 +13,66 @@
 class Client(event.EventsMixin):
     __conn_cls = connection.Connection
 
-    def __init__(self, host, port, *, encoding='UTF-8', ssl=True, loop=None):
+    def __init__(self, host: str, port: str, *, encoding: str = 'UTF-8',
+                 ssl: bool = True, loop: asyncio.BaseEventLoop = None) -> None:
+        """
+        :param host: The IRC server address to connect to.
+        :param port: The port of the IRC server.
+        :param encoding: The character encoding to use.
+        :param ssl: Whether SSL should be used while connecting.
+        :param loop: The event loop to use. Omit unless you know what you're
+        doing.
+        """
         if loop is None:
             loop = asyncio.get_event_loop()
         super().__init__(unpack.parameters, loop=loop)
         self.connection = self.__conn_cls(host, port, self, ssl=ssl,
                                           encoding=encoding, loop=loop)
 
-    def send(self, command, **kwargs):
+    def send(self, command: str, **kwargs) -> None:
         """
         Send a message to the server.
 
-        Examples
-        --------
-        client.send('nick', nick='weatherbot')
-        client.send('privmsg', target='#python', message="Hello, World!")
+        Examples:
+
+        .. code-block:: python
+
+            client.send('nick', nick='weatherbot')
+            client.send('privmsg', target='#python', message="Hello, World!")
 
         """
         packed_command = pack.pack_command(command, **kwargs)
         self.connection.send(packed_command)
 
-    async def connect(self):
+    async def connect(self) -> None:
```

Comment:
So calling a coroutine will return a coroutine object, which means it doesn't really return None.  From the perspective of a user, however, it effectively returns None.

I don't know how this would play with, say, an annotation that expects the same input as to `event_loop.run_until_complete` since None is invalid (but the return from `some_async(*args)` is a valid input.

Basically, is the annotation the return of the body's execution, or the call to the function with args?  These usually align but async changes that a bit.

```python
# Return of async function isn't None, but execution of the body hasn't started
coro = test(arg)
# Object is scheduled, returning the wrapping Task for inspection
task = loop.create_task(coro)
```